tests/test_monitors.py

```diff
@@ -138,6 +148,55 @@ def test_log_monitor_should_pass(make_data):
     assert runner.result.monitor_results[0].error is None
 
 
+def test_warning_monitor_should_fail(make_data):
+    """ WarningCount should fail if the # of warning log messages exceed the limit """
+
+    # Scenario # 1
```

Comment:
**issue**: It is not a good practice to have more than one scenario being tested in the same method. This would make difficult to understand what scenario failed in the test. It will be better to have a separated method for each scenario. It also needs one test for each possible situation (disabled monitor with -1, failing and passing).